dit/utils.py

```diff
@@ -139,8 +139,12 @@ def apply_filter_to(data, date):
     logbook = data.get('logbook', [])
 
     for i in range(len(logbook) + 1):
-        if i == len(logbook) or logbook[-i - 1]['in'] <= date:
+        if i == len(logbook) or logbook[-i - 1]['out'] <= date:
             break
+        elif logbook[-i - 1]['in'] <= date:
+            logbook[-i - 1]['out'] = date
+            break
```

Comment:
This `break` is wrong, remove it. Next items in the logbook may also have 'out' > date.